lib/internal/Magento/Framework/MultiDimensionalIndexer/IndexHandlerInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Framework\MultiDimensionalIndexer;
+
+/**
+ * Represents manipulation with index data
+ *
+ * @api
+ */
+interface IndexHandlerInterface
+{
+    /**
+     * Add data to index
+     *
+     * @param IndexName $indexName
+     * @param \Traversable $documents
+     * @param string $connectionName
+     * @return void
+     */
+    public function saveIndex(IndexName $indexName, \Traversable $documents, string $connectionName);
+
+    /**
+     * Remove given documents from Index
+     *
+     * @param IndexName $indexName
+     * @param \Traversable $documents
+     * @param string $connectionName
+     * @return void
+     */
+    public function cleanIndex(IndexName $indexName, \Traversable $documents, string $connectionName);
```

Comment:
I see, that you use indexName with pre-build information about current dimension - it forse us to build it in client code. But we can pass DImensionInterface (specific for concrete index) which will have all required information for build index table name + the same approach can be use for resolve table name on front

Reply:
Note, this approach we use in draft implementation:  https://github.com/magento-performance/magento2ce/blob/eac69331ff5208ab28ab5ad10db340c3d9674d8a/app/code/Magento/Catalog/Model/Indexer/Product/Price/Shard/ShardTableNameResolver.php#L20https://github.com/magento-performance/magento2ce/blob/eac69331ff5208ab28ab5ad10db340c3d9674d8a/app/code/Magento/Catalog/Model/Indexer/Product/Price/Shard/ShardTableNameResolver.php#L20

Shard - this is a presentation of specific dimensions (sevaral dimenstion can be used, e.g. for Price Indexer)